acceptance/managers/asset_manager.go

```diff
@@ -0,0 +1,313 @@
+)
+
+type AssetManager struct {
+	packPath                    string
+	packFixturesPath            string
+	previousPackPath            string
+	previousPackFixturesPath    string
+	lifecyclePath               string
+	lifecycleDescriptor         builder.LifecycleDescriptor
+	previousLifecyclePath       string
+	previousLifecycleDescriptor builder.LifecycleDescriptor
+	testObject                  *testing.T
+}
+
+func ConvergedAssetManager(t *testing.T, inputConfig InputConfigurationManager) AssetManager {
+	t.Helper()
+
+	var (
+		convergedCurrentPackPath             string
+		convergedPreviousPackPath            string
+		convergedPreviousPackFixturesPath    string
+		convergedDefaultLifecyclePath        string
+		convergedDefaultLifecycleDescriptor  builder.LifecycleDescriptor
+		convergedPreviousLifecyclePath       string
+		convergedPreviousLifecycleDescriptor builder.LifecycleDescriptor
+	)
+
+	assetBuilder := assetManagerBuilder{
+		testObject:  t,
+		inputConfig: inputConfig,
+	}
+
+	if inputConfig.combinations.requiresCurrentPack() {
+		convergedCurrentPackPath = assetBuilder.ensureCurrentPack()
+	}
+
+	if inputConfig.combinations.requiresPreviousPack() {
+		convergedPreviousPackPath = assetBuilder.ensurePreviousPack()
+		convergedPreviousPackFixturesPath = assetBuilder.ensurePreviousPackFixtures()
+
+		h.RecursiveCopy(t, previousPackFixturesOverridesDir, convergedPreviousPackFixturesPath)
+	}
+
+	if inputConfig.combinations.requiresDefaultLifecycle() {
+		convergedDefaultLifecyclePath, convergedDefaultLifecycleDescriptor = assetBuilder.ensureDefaultLifecycle()
+	}
+
+	if inputConfig.combinations.requiresPreviousLifecycle() {
+		convergedPreviousLifecyclePath, convergedPreviousLifecycleDescriptor = assetBuilder.ensurePreviousLifecycle()
+	}
+
+	return AssetManager{
+		packPath:                    convergedCurrentPackPath,
+		packFixturesPath:            currentPackFixturesDir,
+		previousPackPath:            convergedPreviousPackPath,
+		previousPackFixturesPath:    convergedPreviousPackFixturesPath,
+		lifecyclePath:               convergedDefaultLifecyclePath,
+		lifecycleDescriptor:         convergedDefaultLifecycleDescriptor,
+		previousLifecyclePath:       convergedPreviousLifecyclePath,
+		previousLifecycleDescriptor: convergedPreviousLifecycleDescriptor,
+		testObject:                  t,
+	}
+}
+
+func (a AssetManager) PackPaths(kind ComboValue) (packPath string, packFixturesPaths string) {
+	a.testObject.Helper()
+
+	switch kind {
+	case Current:
+		packPath = a.packPath
+		packFixturesPaths = a.packFixturesPath
+	case Previous:
+		packPath = a.previousPackPath
+		packFixturesPaths = a.previousPackFixturesPath
+	default:
+		a.testObject.Fatalf("pack kind must be current or previous, was %s", kind)
+	}
+
+	return packPath, packFixturesPaths
+}
+
+func (a AssetManager) Lifecycle(kind ComboValue) (lifecyclePath string, lifecycleDescriptor builder.LifecycleDescriptor) {
```

Comment:
`LifecycleDescriptor` isn't a path unfortunately. Both of the elements returned in `Pack` are paths. Can you think of a better name to describe return both a path & a descriptor object?

Reply:
You're right. I missed the fact that there are two return values in the `Lifecycle` image (and that the function name is `PackPaths` and not `PackPath` :) ).
What do you think about the name `LifecyclePathAndDesciptor`?
Although it's pretty long, I think it's clearer than just `Lifecycle`.

Reply:
I wouldn't really like such a long method name. The return arguments are named in this method signature which should help alleviate any confusion about what's getting returned.

Reply:
Personally, I have no problem with long method names (even much longer than `LifecyclePathAndDescriptor`). I think it's better to have a long and clear name.
You're right that the return arguments help but I would also change the method name to be more specific. If, theoretically, there would be another function in this file that returns some arguments that relate to `Lifecycle`, how would you change this function name?
Usually, I don't like to prepare code just in case we'll need to make some changes, but in this case I think that the method name is too vague.

Reply:
If there were another lifecycle related thing I'd probably separate this into multiple methods. So I think I should just do that if this isn't clear enough. I'll replace `Lifecycle()` with `LifecyclePath` & `LifecycleDescriptor`

Reply:
I agree with @yaelharel `Lifecycle` is too overloaded in our codebase :)

Reply:
Thank you @simonjjones. Sounds like a good idea!